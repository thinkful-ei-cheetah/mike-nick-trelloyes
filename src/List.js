import React from 'react';
import './List.css';
import Card from './Card';

class List extends React.Component {
  render() {
    const cardsList = this.props.cards.map( (card, index) => 
      <Card key={index} title={card.title} content={card.content} />
    );

    return (
      <section className='List'>
        <header className='List-header'>
          <h2>{this.props.header}</h2>
        </header>
        <div className='List-cards'>{cardsList}</div>
      </section>
    );
  }
};

export default List;

