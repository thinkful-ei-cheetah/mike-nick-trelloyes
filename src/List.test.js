import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List test', () => {
  it('renders List component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List cards={[]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders list of cards', () => {
    const tree = renderer
      .create(
        <List cards={[{ title: 'foo', content: 'bar' }]} header='List1' />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
