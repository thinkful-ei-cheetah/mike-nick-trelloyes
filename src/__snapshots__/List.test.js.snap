// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test renders list of cards 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    List1
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        foo
      </h3>
      <p>
        bar
      </p>
    </div>
  </div>
</section>
`;
